pop11_grammar_ebnf.txt

```diff
@@ -15,6 +16,10 @@ Expression ::=
 	| Label
 	| '(' Sequence ')'
 
+ConstantExpression ::=
+	Construct
+	| '"' Word '"'
+

```

Comment:
We need to distinguish:
- Literals, which only include String, Number i.e. knock-out `'"' Word '"'`, as that is _not_ a literal - it is three tokens that make a constant-expression
- ConstantExpression, which includes the two forms for word literals `'"' Word '"'` and `'"' String '"'`
- Constructs, which is fine as it stands

Reply:
Sure, would this be ok then?
```
ConstantExpression ::=
	Construct
	| '"' Word '"'
	| '"' String '"'

Literal ::=
	Number
	| String
```

Reply:
I'm a little bit confused about Literal since in your email reply yo mentioned this:

> - Adding to Literal the sequence '"' String '"' to accommodate the fact that word-literals can be written as "foo" or "'foo'"

Which is contradictory to what you said here in your comment:

> Literals, which only include String, Number

Reply:
This now looks correct to me. I think the confusion is caused by me trying to comment on it without introducing a new non-Terminal and then adding one a bit later in my review. 

The position that makes sense to me is that a Pop-11 literal is a single token that will evaluate to an immutable value. Tokenisation is well-defined for Pop-11. As such, numbers and strings are tokens but word-expressions are not. (Background: Pop-11 is quite an old programming language and some of its terminology is a bit unusual. Pop-11 words are equivalent to Lisp symbols - a bit like Java's intern'd strings.)

In a normal expression context you denote the symbol _cat_ by using double quotes: `"cat"`. Surprisingly this is _not_ a literal because the quote marks are actually separate tokens. And this genuinely matters inside a ListSegment where the list `[ "cat" ]` would actually generate a list of three elements, each of which was a word. So it makes sense to distinguish literals from what I have called 'constant-expressions'.